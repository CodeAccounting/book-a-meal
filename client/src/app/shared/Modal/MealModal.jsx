import React from 'react';
import PropTypes from 'prop-types';
import Form from '../Form';
import { mealObjPropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @function MealModal
 * @param {object} props
 * @returns {component} AddMealModal
 */
const MealModal = (props) => {
  const { type } = props;
  return <Form {...props} type={type} meta={{ btnText: 'ADD MEAL' }} />;
};

MealModal.propTypes = {
  type: PropTypes.string.isRequired,
  meal: mealObjPropTypes
};

MealModal.defaultProps = {
  meal: null
};

export default MealModal;
