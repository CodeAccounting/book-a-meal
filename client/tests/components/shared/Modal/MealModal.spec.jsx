import React from 'react';
import MealModal from '../../../../src/app/shared/Modal/MealModal';

describe('MealModal', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const shallowWrapper = shallow(<MealModal type="addMeal" />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
