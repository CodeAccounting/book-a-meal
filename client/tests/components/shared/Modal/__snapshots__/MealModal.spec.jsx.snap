// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealModal renders correctly 1`] = `
<Connect(Form)
  meal={null}
  meta={
    Object {
      "btnText": "ADD MEAL",
    }
  }
  type="addMeal"
/>
`;
